pkg/component/runtime/service_command.go

```diff
@@ -93,11 +122,69 @@ func executeCommand(ctx context.Context, log *logger.Logger, binaryPath string,
 }
 
 func executeServiceCommand(ctx context.Context, log *logger.Logger, binaryPath string, spec *component.ServiceOperationsCommandSpec) error {
+	// If there is a previous incarnation of the command executing (due to infinite
+	// retries), make sure to cancel it first before starting a new one here.
+	retryCancelFns.Cancel(binaryPath)
```

Comment:
The command actually needs to be stopped before we can retry it. Is there a way we can confirm that?

https://github.com/elastic/elastic-agent/pull/2226 solved a previous version of this problem, where endpoint would be started before it was stopped and starting the second instance would always fail. There can only be a single instance of endpoint-security running on a machine at one time.

Reply:
You might need to synchronize on the output of the `Wait` call instead, all of the endpoint commands should end eventually there's just no real guarantee about when

https://github.com/elastic/elastic-agent/blob/7772cf2a42bd674a3347e04f9639d1b63ee9aabf/pkg/component/runtime/service_command.go#L77

The start process call is context aware, there's just no guarantee that the process is stopped immediately when the `cancel` function is executed. 

We end up calling `exec.CommandContext` which will call `os.Process.Kill`, but there's a chance it could happen after a small delay.

https://github.com/elastic/elastic-agent/blob/7772cf2a42bd674a3347e04f9639d1b63ee9aabf/pkg/core/process/cmd_darwin.go#L24